In [1]:
import numpy as np
import pandas as pd

In [7]:
df=pd.read_csv('E:\Data Science\Datasets\MegaMart_sales.csv')
df_new=pd.read_csv('E:\Data Science\Datasets\MegaMart_newsales.csv')

In [18]:
df_combined=pd.concat([df,df_new],axis=0,ignore_index=True)
df_combined.groupby(by='Category').Sales.sum()
#df_combined.head()

Category
Furniture          2460
Office Supplies    7970
Technology         8588
Name: Sales, dtype: int64

In [19]:
office_supplies_df=df_combined[df_combined['Category']=='Office Supplies']
off_sup_sales=office_supplies_df['Sales'].sum()
off_sup_sales

7970

#### Combining two data frames
Question 1: Some of the orders are stored in another csv file named `megamart_new sales`. Read the csv file, store it in a data frame and add it to the `megamart_sales` data frame. Find the total sales value of the category 'Office Supplies'  after combining the dataframes

    a)7970
    b)6964
    c)7494
    d)6076

In [ ]:
#your code here


#### Dropping duplicates
Question 2: There are some duplicate rows in the data frame. Drop these rows and calculate the total sales value of the category Office Supplies.

    a)7156
    b)6496
    c)6964
    d)6023

In [21]:
df_combined.drop_duplicates(subset=None,keep='first',inplace=True)
df_combined.groupby(by='Category').Sales.sum()

Category
Furniture          2460
Office Supplies    6964
Technology         7494
Name: Sales, dtype: int64

#### Best category-sub category 
Question 3: Find the most profitable category and sub category combination based on the net profit.


    a)Furniture-Bookcases
    b)Office supplies-Appliances
    c)Office supplies-Storage
    d)Technology-Phones

In [23]:
df_combined.drop_duplicates(subset=None,keep='first',inplace=True)
df_combined.groupby(by=['Category','Sub-Category']).Profit.sum()

Category         Sub-Category
Furniture        Bookcases        308
                 Chairs           -49
                 Furnishings       74
Office Supplies  Appliances       229
                 Art              275
                 Binders          156
                 Envelopes         35
                 Fasteners         10
                 Labels            30
                 Paper             15
                 Storage          192
                 Supplies         103
Technology       Accessories      215
                 Copiers            0
                 Machines         168
                 Phones          1618
Name: Profit, dtype: int64

#### Invalid order IDs
Question 4: How many invalid order IDs are there in the data frame? An order id is of the form AZ-2011-Y where Y represents a whole number. A Order ID is said to be valid only if Y consists of 7 digits. Find the number of invalid order IDs in the data frame.
    
    a)6
    b)7
    c)8
    d)9


In [ ]:
df['Order no']=df['Order ID'].str.split('-')

df['Order no']=df['Order no'].apply(lambda x: x[2])

df['no_of_digits']=df['Order no'].apply(lambda x: len(x))

df['no_of_digits'].value_counts()

#### Occurence of furniture in top 25 sales
Question 5: Find the top 25 orders based on sales value and find the number of orders which belong to furniture category.

    a)2
    b)3
    c)4
    d)5 

In [ ]:
df2=df.sort_values(by='Sales', ascending=False)
df2=df2.iloc[:25,]

df2['Category'].value_counts()

#### And operation
Question 6: Among the orders with sales>250 and profit>50, find the product name of the fourth highest order based on sales value.

    a)Motorola Headset, with Caller ID
    b)Panasonic Printer, Durable	
    c)Hoover Microwave, Red	
    d)Fellowes Lockers, Industrial	

In [ ]:
df3=df.loc[(df['Sales']>250) & (df['Profit']>50)]
df3=df3.sort_values(by='Sales',ascending=False) 

#### Column manipulation
Question 7: Remove the orders with negative profit by dropping the corresponding rows with negative `Profit`. Find the product that makes the lowest profit per Quantity in the Technology category.

    a) Nokia Audio Dock, with Caller ID
    b) Logitech Keyboard, Programmable
    c) Motorola Headset, with Caller ID
    d) Belkin Flash Drive, Bluetooth

In [ ]:
df4=df[df.Profit>0]
df4=df4.loc[df4.Category=='Technology']

df4['Profit_per_qty']=df4['Profit']/df4['Quantity']

df4.sort_values(by='Profit_per_qty',ascending=True)